analytics_data_api/v0/serializers.py

```diff
@@ -96,6 +99,108 @@ class Meta:
             'created'
         )
 
+    def to_representation(self, instance):
+        data = super().to_representation(instance)
+        data['answer_value'] = self._clean_answer_string(data['answer_value'])
+        data['value_id'] = self._clean_answer_string(data['value_id'])
+        return data
+
+    def _clean_answer_string(self, answer_value):
+        """
+        Convert value to a canonical string representation.
+        If value is a list, then returns list values
+        surrounded by square brackets and delimited by pipes
+        (e.g. "[choice_1|choice_3|choice_4]").
+        If value is a string, just returns as-is.
+        If the value contains html the answer_string is parsed as XML,
+        and the text value of the answer_value is returned.
+        """
+
+        def normalize(value):
+            """Pull out HTML tags and deslugify certain escape sequences."""
+            text = self._get_text_from_html(value)
+            return self._parse_slugged_slashes(text)
+
+        # attempt to evaluate string as a list (this will get special
+        # formatting), else keep it a string
+        try:
+            answer_eval = json.loads(answer_value.encode('unicode_escape'))
+            if isinstance(answer_eval, list):
+                answer_value = answer_eval
+        except Exception:  # pylint: disable=broad-except
+            pass
+
+        # produce a string, or 'list' of strings as output
+        # lists containing non-string values such as dictionaries will have
+        # those elements represented as JSON strings
+        if isinstance(answer_value, str):
+            return normalize(answer_value)
+        if isinstance(answer_value, list):
+            normalized = []
+            for value in answer_value:
+                if isinstance(value, str):
+                    normalized.append(normalize(value))
+                else:
+                    normalized.append(json.dumps(value))
+            return u'[{list_val}]'.format(list_val=u'|'.join(normalized))
+
+        return str(answer_value)
+
+    def _get_text_from_html(self, markup):
+        """
+        Convert html markup to plain text.
+        Includes stripping excess whitespace, and assuring whitespace
+        exists between elements (e.g. table elements).
+        """
+        try:
+            root = html5lib.parse(markup)
+            text_list = []
+            for val in self._get_text_from_element(root):
+                text_list.extend(val.strip().split(' '))
+            text = u' '.join(text_list)
+        except Exception:  # pylint: disable=broad-except
+            text = markup.strip()
```

Comment:
I can't seem to create data that would actually cause this function to throw an exception but this is in the old pipeline so I figure we keep this to be safe. Can't hit w/ coverage tho

Reply:
the only possible place for exceptions is html5lib.parse, I can't find a clear description of what exceptions that would throw on a quick reading. If it wants to deal with HTML in the world there is probably no possible malformed HTML that will make it throw an exception.

I'd replace the content here - if something is so horrible that it caused html5lib to choke, we probably should blank it rather than carry it along to transmit over the web to a browser

Reply:
I can get behind just blanking this because you're right, if we end up here something probably gone wrong.